pkg/proxy/conntrack/conntrack.go

```diff
@@ -57,8 +58,12 @@ func newConntracker(handler netlinkHandler) Interface {
 }
 
 // ListEntries list all conntrack entries for connections of the given IP family.
-func (ct *conntracker) ListEntries(ipFamily uint8) ([]*netlink.ConntrackFlow, error) {
-	return ct.handler.ConntrackTableList(netlink.ConntrackTable, netlink.InetFamily(ipFamily))
+func (ct *conntracker) ListEntries(ipFamily uint8) (entries []*netlink.ConntrackFlow, err error) {
+	util.RetryOnIntr(func() error {
+		entries, err = ct.handler.ConntrackTableList(netlink.ConntrackTable, netlink.InetFamily(ipFamily))
+		return err
+	})
+	return entries, err
```

Comment:
So in the `EINTR` case, the list function still potentially returns a _partial_ list.

For `CleanStaleEntries`, if it gets back `EINTR` and a partial list from `ct.ListEntries`, it can still clean up any stale entries that appeared in that partial list, which is a better result than if it just throws the partial list away. So you should update `CleanStaleEntries` so if it gets back `EINTR` here, it still logs an error, but then continues trying to do cleanup.

(You should also update the `ListEntries` doc comment to mention that it can return a partial list and an error.)

Reply:
Gotcha, I've handled that. I'll double check if this applies to other places too